tool/tsh/db.go

```diff
@@ -413,6 +425,34 @@ func needRelogin(cf *CLIConf, tc *client.TeleportClient, database *tlsca.RouteTo
 	return mfaRequired, nil
 }
 
+// dbUserHasChanged checks if cf.DatabaseUser username matches the username in the database certificate.
+func dbUserHasChanged(cf *CLIConf, certPath string) (bool, error) {
```

Comment:
Nit: Maybe rename to something like `dbInfoHasChanged` since it now checks both user and database names?
```suggestion
func dbInfoHasChanged(cf *CLIConf, certPath string) (bool, error) {
```